tasks/libs/common/git.py

```diff
@@ -71,17 +71,19 @@ def get_file_modifications(
 
     flags = '--no-renames' if no_renames else ''
 
-    modifications = [
+    raw_modifications = [
         line.split('\t')
         for line in ctx.run(f"git diff --name-status {flags} {last_main_commit}", hide=True).stdout.splitlines()
     ]
-
     if added or modified or removed:
-        modifications = [
-            (status, file)
-            for status, file in modifications
-            if (added and status == "A") or (modified and status in "MCRT") or (removed and status == "D")
-        ]
+        modifications = []
+        for m in raw_modifications:
+            if len(m) == 3:
+                # file was renamed
+                continue
+            status, file = m
+            if (added and status == "A") or (modified and status in "MCRT") or (removed and status == "D"):
+                modifications.append((status, file))
 
     if only_names:
         modifications = [file for _, file in modifications]
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
Is it fine to have this change of the logic here? Before this was taking current `raw_modifications` when `not added and not modified and not removed`, now it's taking a non defined array

Reply:
Good catch. Code updated.